omod/src/test/java/org.openmrs.module.smartonfhir/web/servlet/SmartLaunchOptionSelectedTest.java

```diff
@@ -0,0 +1,121 @@
+	
+	private static final String TOKEN_PATIENT = "http://localhost:8180/auth/realms/openmrs/login-actions/action-token?key=eyJhbGciOiJIUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJkZjJhYTBjMC0wMzFjLTQ5ZjItYTczMC1hYmM0Mjg2OTM1Y2UifQ.eyJleHAiOjE2NDEyNzIyNzIsImlhdCI6MTY0MTI3MTk3MiwianRpIjoiMjIwZDlmYWMtYjgyNi00ZjVhLWJmM2UtNDg4NmY3MjI3OWFiIiwiaXNzIjoiaHR0cDovL2xvY2FsaG9zdDo4MTgwL2F1dGgvcmVhbG1zL29wZW5tcnMiLCJhdWQiOiJodHRwOi8vbG9jYWxob3N0OjgxODAvYXV0aC9yZWFsbXMvb3Blbm1ycyIsInN1YiI6ImY6ZDllNTk0ZDItMWE5NC00YjNjLTkwZTQtODBhYmI5ODAzOTY4OjEiLCJ0eXAiOiJzbWFydC1wYXRpZW50LXNlbGVjdGlvbiIsIm5vbmNlIjoiMjIwZDlmYWMtYjgyNi00ZjVhLWJmM2UtNDg4NmY3MjI3OWFiIiwiYXNpZCI6IjcwZmU4ODNjLWVhNGUtNGRmNy1iYmI5LWU0MTdjNGVlOWI5OS5JOUlhSEFGWFBaMC45MzBkM2JkYS1iNTY4LTQ1YTItODgyZS05MGQxMTNjNjMxNzciLCJhc2lkIjoiNzBmZTg4M2MtZWE0ZS00ZGY3LWJiYjktZTQxN2M0ZWU5Yjk5Lkk5SWFIQUZYUFowLjkzMGQzYmRhLWI1NjgtNDVhMi04ODJlLTkwZDExM2M2MzE3NyIsInVzZXIiOiJleUpoYkdjaU9pSklVekkxTmlJc0luUjVjQ0lnT2lBaVNsZFVJbjAuZXlKbGVIQWlPakUyTkRFeU56SXlOeklzSW1semN5STZJbWgwZEhBNkx5OXNiMk5oYkdodmMzUTZPREU0TUM5aGRYUm9MM0psWVd4dGN5OXZjR1Z1YlhKeklpd2lZWFZrSWpvaWFIUjBjRG92TDJ4dlkyRnNhRzl6ZERvNE1EZ3dJaXdpYzNWaUlqb2lZV1J0YVc0aUxDSjBlWEFpT2lKemJXRnlkQzExYzJWeWJtRnRaUzEwYjJ0bGJpSXNJbUZ6YVdRaU9pSnpiV0Z5ZEVOc2FXVnVkQ0o5LnFoeHE1Z3Z1RERFcmNUbFYyY0lzOFpLSHpzd2VoZWkxRVlqZDNJR05mTk0iLCJsYXVuY2hUeXBlIjoiL3BhdGllbnQifQ.7rqOafcWeQhr0YGhG6cxrIrKUarDTxl3lb0dBi-Hjkk&client_id=smartClient&tab_id=Ud8TKlvjbh4&execution=9e89e1f3-41cd-4d92-946c-d8f56e9af62a&app-token=%7BAPP_TOKEN%7D";
+	
+	private static final byte[] SMART_SECRET_KEY_HOLDER = "SecretKey".getBytes(StandardCharsets.UTF_8);
+	
+	private static final String PATIENT_UUID = "12345";
+	
+	private static final String VISIT_UUID = "56789";
+	
+	private MockHttpServletRequest request;
+	
+	private MockHttpServletResponse response;
+	
+	private SmartLaunchOptionSelected smartLaunchOptionSelected;
+	
+	@Before
+	public void setup() throws Exception {
+		request = new MockHttpServletRequest();
+		response = new MockHttpServletResponse();
+		smartLaunchOptionSelected = new SmartLaunchOptionSelected();
+		
+		mockStatic(SmartSecretKeyHolder.class);
+		
+		doReturn(SMART_SECRET_KEY_HOLDER).when(SmartSecretKeyHolder.class, "getSecretKey");
+	}
+	
+	@Test
+	public void shouldReturnCorrectURLWhenLaunchTypeIsEncounter() throws IOException {
+		request.setParameter("patientId", PATIENT_UUID);
+		request.setParameter("token", TOKEN_ENCOUNTER);
+		
+		smartLaunchOptionSelected.doGet(request, response);
+		
+		assertThat(response, notNullValue());
+		assertThat(response.getRedirectedUrl(), notNullValue());
+		assertThat(response.getRedirectedUrl().contains("app=smartonfhir.search.visit"), equalTo(true));
+		assertThat(response.getRedirectedUrl().contains("patientId=12345"), equalTo(true));
```

Comment:
These will give better error messages:

```suggestion
		assertThat(response.getRedirectedUrl(), containsString("app=smartonfhir.search.visit"));
		assertThat(response.getRedirectedUrl(), containsString("patientId=12345"));
```

Reply:
This same change should be applied elsewhere